op-batcher/batcher/driver.go

```diff
@@ -465,68 +505,53 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 
 			if blocksToLoad != nil {
 				// Get fresh unsafe blocks
-				if err := l.loadBlocksIntoState(l.shutdownCtx, blocksToLoad.start, blocksToLoad.end); errors.Is(err, ErrReorg) {
+				if err := l.loadBlocksIntoState(ctx, blocksToLoad.start, blocksToLoad.end); errors.Is(err, ErrReorg) {
 					l.Log.Warn("error loading blocks, clearing state and waiting for node sync", "err", err)
 					l.waitNodeSyncAndClearState()
 					continue
+				} else {
+					l.sendToThrottlingLoop(pendingBytesUpdated) // we have increased the pending data. Signal the throttling loop to check if it should throttle.
+					l.signalPublishingLoop(blocksLoadedCh)      // signal the write loop that blocks have been loaded
 				}
```

Comment:
because of the `continue`, you could move this out of the `else` branch, like the publish call before.

Reply:
69858c97b5633b3ffba62d159664a47f727ead1c